### ISSUE TITLE
Implement selected set of SKOS annotations in place of rdfs:comment and rdfs:label

### ISSUE THREAD

**Issue by rjyounes** [semanticarts/gist#351]
Depends on #85.

**Comment by rjyounes** [comment]
Multiple assignees could work on the same branch.

**Comment by uscholdm** [comment]
Hmm, normally I like to only work on a branch on my own, but in this case it makes a lot of sense -- so long as people agree up front on what definitions they will work on - to avoid clashes and duplication of effort.

**Comment by rjyounes** [comment]
This is a common practice in software development, at least in my experience. I'm assuming upfront agreement, which would also be necessary if we used different branches.

**Comment by rjyounes** [comment]
@uscholdm I forgot to mention the reason for working on one branch: this is an atomic change, and given that every version merged to develop should be well-formed, we don't want some of the updated definitions but not all to be merged to develop in a different pull request. A different scenario - perhaps better - is that we would each create a branch off of the main feature branch rather than off of develop. Those branches would each get merged into the higher branch before that gets merged to develop.

**Comment by uscholdm** [comment]
Are you saying to create a main branch off of develop for the annotation work,  but nobody does work directly on that branch, but on sub-branches.  SO it acts as a develop branch for the annotations work.  That is a good idea.

**Comment by rjyounes** [comment]
Yes, exactly. Then everything gets merged to develop at once.

**Comment by rjyounes** [comment]
We did not explicitly make a decision about rdfs:label. I assume we want to replace rdfs:label with skos:prefLabel.

**Comment by rjyounes** [comment]
This should include corresponding additions to the gist style guide.

**Comment by rjyounes** [comment]
Step 1: Global search and replace. 
Step 2: Parse prefLabel into scopeNote, example, etc.

**Comment by rjyounes** [comment]
Will raise in an email to see who has applications that would be impacted. Ask recipients to post a comment to this issue. Meika will work with Melissa to send the email.

**Comment by JonathonGist** [comment]
I will be absolutely delighted if this change is implemented! If it is implemented, it will have no deleterious affects whatsoever for me.

**Comment by tedhills** [comment]
In general, I agree with the thrust of this effort to replace rdfs:comment with more specific annotations, but the devil is in the details, and I need more details.

It's not clear to me what the exact list of changes is that is planned to be implemented. Is it what was specified by Issue #85? That issue had a debate over rdfs:seeAlso that was unresolved.

Please list the annotations to be used and their planned usage. Thanks!

**Comment by rjyounes** [comment]
@tedhills You are correct, the changes are identified in #85.

**Comment by conceptminers1** [comment]
Presently I do not know the exact use cases to exemplify my agreement but I think the changes made would expand the GIST and allow for including SKOS related use cases, making GIST more acceptable in varied domains. Wishing Best.

**Comment by vberkel** [comment]
The replacement of rdfs:comment makes sense, more precise terminology and negligible impact on our systems. 

The replacement of rdfs:label does have some impact (relative to changing URIs schemes it is a low amount) on our systems. What is the benefit of switching if you are not including the skos:altLabel and skos:hiddenLabel as well?  I do not see in the related github issues a mention of these.

**Comment by tedhills** [comment]
@rjyounes Thanks for the list. It all looks good and I support the changes as beneficial.

In issue #85, there is some discussion of rdfs:seeAlso. I suggest that rdfs:seeAlso be added to the list of annotations that replace rdfs:comment, wholly or partially (usually partially in this case). I have used rdfs:seeAlso only to reference other IRIs in the same ontology, but I see from reading [the seeAlso spec](https://www.w3.org/TR/rdf-schema/#ch_seealso) that the object of the annotation property could be any IRI, including references to non-OWL resources, such as documentation Web pages. Thoughts?

**Comment by rjyounes** [comment]
@tedhills I like the addition of rdfs:seeAlso, and I agree that it can point to a resource or just a URL. @vberkel Agreed, we should add skos:altLabel as well. Personally I do not like skos:hiddenLabel since it is used application-oriented rather than purely semantic. I am going to reopen #85 and add these for discussion during our next review meeting. Thanks for the input.

**Comment by uscholdm** [comment]
I don't think we need `rdfs:seeAlso`. Here is why: 

`skos:scopeNote` is is an instance of `owl:AnnotationProperty` (and by inference, of `rdf:Property`). It "helps to clarify the meaning and/or the use of a concept."

`rdfs:seeAlso` is an instance of `rdf:Property` that "_is used to indicate a resource that might provide additional information about the subject resource_". It is not an `owl:AnnotationProperty`, but it can take any `rdf:Resource` as a value (which includes `rdf:Literal`).

`skos:note` is is an instance of `owl:AnnotationProperty`. It means "_A general note, for any purpose._"

`skos:note` is nearly identical in meaning to `rdfs:seeAlso`, and perhaps should have been modeled as a subproperty of  `rdfs:seeAlso` show the connection.

For uniformity's sake, I prefer to just stick with skos annotations.  `rdfs:seeAlso` does not add anything that I can see.

See: https://www.w3.org/TR/rdf-schema/#ch_seealso,

**Comment by rjyounes** [comment]
skos:note is the superproperty of the other skos note annotations - scopeNote, historyNote, etc. As such it is (human) semantically vacuous and this is why we chose not to use it. rdfs:seeAlso has specific (human) semantics that are different - it specifically references another resource. So from that point of view I don't see them as identical, nor do I see that it ought to be a subproperty of rdfs:seeAlso. 

rdfs:isDefinedBy is a subproperty of rdfs:seeAlso, so given that we are using and automatically populating the former, I don't see why we would object to the latter.

**Comment by uscholdm** [comment]
`skos:note` is no more human vacuous in meaning than `rdfs:seeAlso`. 
- `rdfs:seeAlso`: "is used to indicate a resource that might provide additional information about the subject resource"
- `skos:note`: "A general note, for any purpose."

If each implies the other, they are equivalent. I think this holds.
- Anything that provides additional information can be seen as a note (for some purpose)
- Anything that is a note for any purpose is necessarily there to provide additional information about the subject.

**Comment by uscholdm** [comment]
The only reason to use 'rdfs:seeAlso' is if there is not a more specific subproperty of it that is a good fit (e.g. rdfs:definedBy).
For the same reason, one might wish to use skos:note. 

If we prohibit use of skos:note, then why do want to allow use of rdfs:seeAlso?

The alternative in both cases is to create a new property.

**Comment by Jamie-SA** [comment]
I disagree that they are equivalent. rdfs:seeAlso is a reference to something else, skos:note is self contained. Two very different meanings.

In addition, if the object of a rdfs:seeAlso triple is a URI that URI can be machine actionable (resolvable, self describing, etc), skos:note is not.

**Comment by Jamie-SA** [comment]
In #85 @uscholdm states that rdfs:seeAlso does not add anything. I disagree with that, it is the only term (annotation) we are currently looking at that specifically refers to something else to provide additional information.

**Comment by uscholdm** [comment]
@Jamie-SA says that `rdfs:seeAlso` "specifically refers to something else to provide additional information"

The something else (according to the range of the property) is always a` rdf:Resource`, which includes `rdf:Literal`. According to that, here is a legitimate use:
- `gist:Intention rdfs:seeAlso "What one intends to do"^^xsd:string`

While there is nothing to indicate this in the documentation, it could be that the intended use is for the object (whether a URL or literal) to be a _pointer_ to something else.  That would make sense and would distinguish it from `skos:note`.  We could decide to use `rdfs:seeAlso` this way.

**Comment by rjyounes** [comment]
@uscholdm The difference you're missing between skos:note and rdfs:seeAlso is that the latter specifically points to _another resource_. skos:note is just a note, and as an owl:DatatypeProperty can contain only text.

**Comment by rjyounes** [comment]
This discussion should be moved over to #85. We are carrying it on in two places, and it rightfully belongs there, which is a discussion of which properties to adopt, as opposed to here, which assumes a decision on which properties to adopt has been made and is concerned only with implementation.

**Comment by rjyounes** [comment]
Setting back to in review because new questions about the set of annotations have been raised (#85).

**Comment by rjyounes** [comment]
See #85 for final decision.

**Comment by rjyounes** [comment]
Initially, automation:

- rdfs:label -> skos:prefLabel
- rdfs:comment -> skos:definition - but needs human review

For backward compatibility, generate a file with the rdfs annotations for the release package. Release notes should also specify a query for users to make the update in their data.

Human intervention is required for:
- breaking up skos:definition into multiple annotations, where relevant

From develop: create branch feature/skos_annotations
Then create: feature/skos_annotations/rebecca
feature/skos_annotations/michael
etc.
Each developer must merge changes from skos_annotations into their branch before making a PR to skos_annotations.
Once all PRS are merged to skos_annotations, someone submits a PR to merge skos_annotations into develop.

Automation to generate rdfs:label/comment file: Boris will add to ontology-toolkit. Has created issue and assigned to himself.
Michael: Do automated search as above and then manual review/edits
Ask at least 3 people to review. 

Rebecca: create separate issues

**Comment by rjyounes** [comment]
Final decision from #85 is repeated here:

Stop using `rdfs:comment` and `rdfs:label`. Instead use:

#### Labels

`skos:prefLabel`
`skos:altLabel`

#### Comments

`skos:definition`
`skos:example`
`skos:scopeNote`
`skos:editorialNote` - just for future editors, doesn't affect users' understanding
`rdfs:isDefinedBy` (automatically inserted by ontology-toolkit)
`rdfs:seeAlso` (can be URI or literal)


#### Limited use - style guide should define appropriate circumstances.

`skos:historyNote` (occasionally for things that have been around for a long time)
`skos:changeNote` (rarely, since should be in release notes, but users are not always looking at release notes. Use, e.g., to describe rationale)
`skos:not` (rarely if ever, if no more specific note is available and not want to create yet another annotation property)

**Comment by rjyounes** [comment]
This is a patch update because we will provide the additional file containing the old RDFS annotations.

**Comment by rjyounes** [comment]
@uscholdm 
- Coordinate on issue #377 - they should be merged to develop for the same release.
- Assign a minimum of 3 reviewers. (@rjyounes would like to be one of them.)

**Comment by uscholdm** [comment]
After this ic completed, issue https://github.com/semanticarts/gist/issues/379 must be taken care of.
